modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/service/test/DLFolderServiceTest.java

```diff
@@ -97,6 +98,34 @@ public void setUp() throws Exception {
 		_alternativeGroup = GroupTestUtil.addGroup();
 	}
 
+	@Test
+	public void testAddDLFolderWithExternalReferenceCode() throws Exception {
+		String externalReferenceCode = RandomTestUtil.randomString();
+
+		Folder folder = addFolder(externalReferenceCode);
+
+		Assert.assertEquals(
+			externalReferenceCode, folder.getExternalReferenceCode());
+	}
+
+	@Test
+	public void testAddDLFolderWithoutExternalReferenceCode() throws Exception {
+		Folder folder = addFolder(null);
+
+		Assert.assertEquals(
+			folder.getExternalReferenceCode(),
+			String.valueOf(folder.getUuid()));
```

Comment:
Is the `valueOf` as the uuid is already a String?

Reply:
good catch!